### PR TITLE
RF: Move all patool code into a dedicated code file

### PR DESCRIPTION
**Pull request by mih**
This shall make providing an alternative but API compatible
implementation easier to place and test alongside of this existing
one.

Towards #4019

### CODE REVIEW

**Comment by mih** [comment]
I will ignore the coverage report. Moving untested code doesn't make things substantially worse.

File: datalad/tests/test_archives.py [hunk 1]
```diff
@@ -25,8 +25,8 @@
     compress_files,
     decompress_file,
     ExtractedArchive,
-    unixify_path,
 )
+from datalad.support.archive_utils_patool import unixify_path
```
Comment (yarikoptic) [inline datalad/tests/test_archives.py]: Why to introduce absolute import when the rest (at least below) are relative?

File: datalad/support/archive_utils_patool.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+import os
+from .exceptions import MissingExternalDependency
+from .path import (
+    basename,
+    join as opj,
+    exists,
+)
+
+from datalad.utils import (
```
Comment (yarikoptic) [inline datalad/support/archive_utils_patool.py]: While at it make into a relative import? Note that there are more imports from ..utils below

**Comment by mih** [comment]
Thx for the review!

re imports: Everything is switched to absolute. If you check the follow-up PR, you will see that no relative imports are left. If you are concerned about this intermediate state, we can leave this hanging here, or even close it till #4041 is done. I just wanted to disentangle to move from the subsequent changes. Your call, I am fine either way.

**Comment by yarikoptic** [comment]
I am confused on where and why everything changed to absolute, since I thought ideally we should use relative, but ok, I guess we could RF later.

**Pull request merged**